### ISSUE TITLE
Multiple type params raise a warning

### ISSUE THREAD

**Issue by m-sund** [Luracast/Restler#637]
Using version 3.1.0.

All multiple type params raise a warning to error log.

`PHP message: PHP Warning:  Illegal offset type in isset or empty in .../Luracast/Restler/Data/Validator.php on line 427`

Example of a multiple params case:

```
     * @param string|boolean  $a                 Param A
     * @param string|int  $b                 Param B
```
The cause is this line:

`isset(static::$preFilters[$info->type]) &&`

where the $info->type is actually an array because of the way handling multiple params in the CommentParser.php (string is exploded with '|')

Has anyone solved this issue already?

**Comment by m-sund** [comment]
Quick fix was to add !is_array() to if clause to get rid of warnings. Yet I am not sure what else issues $info->type as an array causes in Restler.

```diff --git a/vendor/luracast/restler/vendor/Luracast/Restler/Data/Validator.php b/vendor/luracast/restler/vendor/Luracast/Restler/Data/Validator.php
index 023fc7c4..5aebe495 100644
--- a/vendor/luracast/restler/vendor/Luracast/Restler/Data/Validator.php
+++ b/vendor/luracast/restler/vendor/Luracast/Restler/Data/Validator.php
@@ -424,6 +424,7 @@ class Validator implements iValidate
             $input = $func($input);
         }
         if (
+            !is_array($info->type) &&
             isset(static::$preFilters[$info->type]) &&
             (is_scalar($input) || !empty($info->children)) &&
             is_callable($func = static::$preFilters[$info->type])
```

**Comment by Arul-** [comment]
Restler does not support multi-type parameters. Anyway we applied your suggestion
Fixed applied in c50fff21cf546ca76bd1935f226543bbc2d7d006 now available in v5 branch

Sorry about the delay

**Comment by Arul-** [comment]
Now available in 5.0.7
